### PR TITLE
Naming: Hide providers deployers and wrappers

### PR DESCRIPTION
**Pull request by maraoz**
With this PR, `FooWrapper`s are now called `Foo`, and contain the `FooDeployer` and `FooProvider` by composition (via the `deploy()` and `fetch()` static functions)

Related to https://github.com/zeppelinos/zos-lib/issues/68

### CODE REVIEW

File: src/app/App.js [hunk 1]
```diff
@@ -17,8 +20,16 @@ export default class AppWrapper {
     this.txParams = txParams
   }
 
+  static async fetch() {
+    return await AppProvider.from.apply(AppProvider, arguments);
```
Comment (spalladino) [inline src/app/App.js]: `AppProvider.from(...arguments)` [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) is cleaner here (assuming it's supported by the current babel presets)

Comment (spalladino) [reply src/app/App.js]: Really like having fetch and deploy available here btw

File: src/release/Release.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ *
+ */
+export default class Release {
+  constructor(release, txParams = {}) {
+    this._release = release
+    this.txParams = txParams
+  }
+
+  static async deploy() {
```
Comment (spalladino) [inline src/release/Release.js]: I'd add a `fetch` method for completeness

File: src/app/App.js [hunk 3]
```diff
@@ -17,8 +20,16 @@ export default class AppWrapper {
     this.txParams = txParams
   }
 
+  static async fetch() {
```
Comment (facuspagnuolo) [inline src/app/App.js]: what about using `at` here? IMO it is already adopted by the community

File: src/app/App.js [hunk 4]
```diff
@@ -17,8 +20,16 @@ export default class AppWrapper {
...
+    return await AppProvider.from(...arguments);
+  }
+
+  static async deploy() {
+    return await AppDeployer.deploy(...arguments);
+  }
```
Comment (facuspagnuolo) [inline src/app/App.js]: Nit: I'd move static methods above the constructor

File: src/package/Package.js [hunk 5]
```diff
@@ -1,13 +1,28 @@
...
 
+import PackageDeployer from './PackageDeployer'
+import PackageProvider from './PackageProvider'
+
 const log = new Logger('Package')
 
-export default class PackageWrapper {
+/**
+ *
+ */
```
Comment (facuspagnuolo) [inline src/package/Package.js]: (?)

File: src/package/Package.js [hunk 6]
```diff
@@ -1,13 +1,28 @@
...
     this.txParams = txParams
   }
 
+  static async fetch() {
+    return await PackageProvider.from(...arguments);
+  }
+
+  static async deploy() {
+    return await PackageDeployer.deploy(...arguments);
+  }
```
Comment (facuspagnuolo) [inline src/package/Package.js]: same here

File: src/app/App.js [hunk 7]
```diff
@@ -17,8 +20,16 @@ export default class AppWrapper {
     this.txParams = txParams
   }
 
+  static async fetch() {
+    return await AppProvider.from(...arguments);
+  }
+
+  static async deploy() {
```
Comment (facuspagnuolo) [inline src/app/App.js]: and what do you think about using `new` here?

File: src/app/App.js [hunk 8]
```diff
@@ -4,11 +4,14 @@ import Logger from '../utils/Logger'
...
+import AppDeployer from './AppDeployer'
+
 const log = new Logger('App')
 
-export default class AppWrapper {
-  constructor(app, factory, appDirectory, _package, version, txParams = {}) {
-    this.app = app
+export default class App {
+  constructor(_app, factory, appDirectory, _package, version, txParams = {}) {
+    this._app = _app
```
Comment (spalladino) [inline src/app/App.js]: Something I kept thinking after the original review: why hiding the wrapped App contract now, when we are still exposing all dependent contracts (package, factory, and directories)? On the other hand, the inner contract does not seem to be used anywhere (just checked the CLI), so it'd be ok to remove.

**Comment by facuspagnuolo** [comment]
@maraoz we are merging this PR since we need some of your changes, please review comments later and address them if needed.

**Pull request merged**